### ISSUE TITLE
[QUESTION] why the video send packet thread codes works on viewer but not on master with webrtc test page

### ISSUE THREAD

**Issue by ajjiang48** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1153]
Hi, Sir,
     I'm working on our multi-camera dash cam device integrating the webrtc (using C SDK) video streaming; I first implement the webrtc viewer in the dash cam; the video codec is H264 and no audio; it works; tested with webrtc test page (as master), and then, I implement the webrtc master in the dash cam, and use the same video-packet-send codes and test it with webrtc test page (as viewer), but it's not working (I see the video -frame is sent out but it's not displayed in the test page's "from master" video box.

     I've saved both the verbose-level logs as the below attachment (session_0521_master.log is the dash cam as master, test page as viewer (the failure case), session_0525_viewer.log is the dash cam as viewer, test page as master (the OK case)).
     I notice that in the failure case (dash cam as master; test page as viewer), I can see many rtcpReportsCallback function log with 0 pakcets 0 bytes => 
"VERBOSE rtcpReportsCallback(): sender report 579844362 16452218027177847908 77836402214477 : 0 packets 0 bytes".
however, in the OK log, there's no rtcpReportsCallback log with 0 packets 0 bytes.

     so, as for the above explanation and logs, my question is : the video sending thread in my viewer and master implementation is identical and why the master implementation NOT working ??

Thanks,
A.J.

[session_0521_master.log](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/files/6547761/session_0521_master.log)
[session_0525_viewer.log](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/files/6547764/session_0525_viewer.log)

**Comment by disa6302** [comment]
@ajjiang48 ,

I belive you are using the GStreamer sample?
Any changes you have made to the sample apart from the pipeline?

**Comment by ajjiang48** [comment]
@disa6302 I do this from the viewer then the master; and I just send the H264 video packets out and NOT touch the incoming frame, so, I just use the kvsWebRTCClientMaster.c sample; I believe the key is the video-sending thread (for both master and viewer, after done the SDP handshake, bring-up the video-sending thread and keep sending the video frame).
I could paste my video-sending thread code if needed.

Thanks,
A.J.

**Comment by disa6302** [comment]
Can you run the sample as is and see if it works?

**Comment by ajjiang48** [comment]
@disa6302 yes, I do build the  kvsWebRTCClientMaster.c sample and run on the dash cam device and test with test page.; it's working.

**Comment by disa6302** [comment]
@ajjiang48 ,

In that case, I would suggest you to debug the changes in your application to backtrack what could be causing the issue

**Comment by ajjiang48** [comment]
@disa6302 I will check that later; could you help check the 2 logs I provided and why the rtcpReportsCallback function with 0 packets and 0 bytes often appears in the failed case  (master implementation) ??

**Comment by disa6302** [comment]
That is because viewer is not receiving anything from master

**Comment by disa6302** [comment]
I am closing this issue because your application requires debugging on your end. Feel free to reopen if you have any questions after debugging

**Comment by ajjiang48** [comment]
@disa6302 it's not always 0 packets 0 bytes; it's interleaved like the below:

2021-05-21 10:08:32 VERBOSE rtcpReportsCallback(): sender report 579844362 16452218027177847908 77836402214477 : 0 packets 0 bytes
2021-05-21 10:08:32 VERBOSE rtcpReportsCallback(): sender report 207970803 16452218027724431030 225602 : 66 packets 25128 bytes
2021-05-21 10:08:33 VERBOSE rtcpReportsCallback(): sender report 579844362 16452218028193272666 77836402225825 : 0 packets 0 bytes
2021-05-21 10:08:33 VERBOSE rtcpReportsCallback(): sender report 207970803 16452218028791810718 247968 : 90 packets 47904 bytes
2021-05-21 10:08:33 VERBOSE rtcpReportsCallback(): sender report 579844362 16452218029217641693 77836402237274 : 0 packets 0 bytes
2021-05-21 10:08:33 VERBOSE rtcpReportsCallback(): sender report 207970803 16452218029541251588 263673 : 93 packets 48339 bytes
2021-05-21 10:08:33 VERBOSE onRtcpSenderReport(): RTCP_PACKET_TYPE_SENDER_REPORT 2086657781 16452218255918350722 rtpTs: 3050995622 148 pkts 10912 bytes
